src/status_im/ethereum/json_rpc.cljs

```diff
@@ -0,0 +1,74 @@
+(ns status-im.ethereum.json-rpc
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [status-im.ethereum.decode :as decode]
+            [status-im.native-module.core :as status]
+            [status-im.utils.ethereum.abi-spec :as abi-spec]
+            [status-im.utils.money :as money]
+            [status-im.utils.types :as types]
+            [taoensso.timbre :as log]))
+
+(def json-rpc-api
+  {"eth_call" {}
+   "eth_getBalance"
+   {:on-result money/bignumber}
+   "eth_estimateGas"
+   {:on-result money/bignumber}
+   "eth_gasPrice"
+   {:on-result money/bignumber}
+   "eth_getBlockByHash"
+   {:on-result #(-> (update % :number decode/uint)
+                    (update :timestamp decode/uint))}
+   "eth_getTransactionByHash" {}
+   "eth_getTransactionReceipt" {}
+   "eth_newBlockFilter" {:subscription? true}
+   "eth_newFilter" {:subscription? true}})
+
+(defn call
+  [{:keys [method params on-success on-error]}]
+  (when-let [method-options (json-rpc-api method)]
+    (let [{:keys [id on-result subscription?]
+           :or {on-result identity
+                id        1
+                params    []}} method-options
+          on-error (or on-error
+                       #(log/error :json-rpc/error method :params params :error %))]
+      (if (nil? method)
+        (log/error :json-rpc/method-not-found method)
+        (status/call-private-rpc
```

Comment:
it would be good to have something like `(status/call-private-rpc "name" params)`

Reply:
That is what I made initially but it turns out you always have different options so I figured it is better to have this generic function taking a map.